File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int res = 1;
        Map<String, Integer> map = new HashMap<>();
        for(String[] cloth : clothes) {
            map.put(cloth[1], map.getOrDefault(cloth[1], 0) + 1);
        }
        Set<String> keys = map.keySet();
        for(String key : keys){
            res *= map.get(key)+1;
        }
        
        return res - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let res = 1;
    const map = new Map();
    for(let i=0; i<clothes.length; i++){
        const key = clothes[i][1];
        map.set(key, (map.get(key) || 0) + 1);
    }
    
    
    map.forEach((val, key) => {
        res *= val + 1;
    });
    
    
    
    return --res;
}